File: src/ProjectR/content/scripts/spells/GrandHealingPotion.lua
function GetTargetType()
    return Single
end

function GetName()
return "Grand Healing Potion"
end

function GetDescription()
return "Heals target for 50%% of their max. HP."
end

function IsSupportSpell()
    return true
end

function GetMPCost()
    return .88
end

function GetDelay()
  return .25
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect() 
  d:Heal(dHP / 2)
end

File: src/ProjectR.Scripting/content/scripts/spells/GrandHealingPotion.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class GrandHealingPotion : SpellScriptBase
    {
        public override string Name { get { return "Grand Healing Potion"; } }
        public override string Description { get { return "Heals target for 50%% of their max. HP."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return true; } }

        public override double MPCost { get { return 88; } }
        public override double Delay { get { return .25; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            Heal(tHP / 2);
        }
    }
}